bVNC/src/main/java/com/undatech/opaque/ConnectionGridActivity.java

```diff
@@ -514,13 +524,52 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return true;
     }
 
-    public void onTextObtained(String dialogId, String[] obtainedStrings, boolean wasCancelled, boolean keep) {
+    public void onTextObtained(String dialogId, String[] obtainedStrings, boolean wasCancelled, boolean keep, boolean[] obtainedBooleans) {
         Log.i(TAG, "onTextObtained");
-        handlePassword(obtainedStrings[0], wasCancelled);
+        if (dialogId == GetTextFragment.DIALOG_ID_GET_MATCHING_MASTER_PASSWORDS && !wasCancelled)  {
+            Utils.setSharedPreferenceBoolean(this, Constants.onlyLockConnectionEditing, obtainedBooleans[0]);
+        }
+        handlePassword(obtainedStrings[0], wasCancelled, dialogId,
+            Utils.querySharedPreferenceBoolean(this, Constants.onlyLockConnectionEditing)
+        );
     }
 
-    public void handlePassword(String providedPassword, boolean dialogWasCancelled) {
+    public void handlePassword(String providedPassword, boolean dialogWasCancelled, String dialogId, boolean onlyLockEditingConnections) {
         Log.i(TAG, "handlePassword");
+        boolean loadConnections = onlyLockEditingConnections ? handlePasswordOnlyLockEditingConnections(providedPassword, dialogWasCancelled, dialogId) : handlePasswordDatabaseLock(providedPassword, dialogWasCancelled);
+        if (loadConnections) {
+            removeGetPasswordFragments();
+            loadSavedConnections();
+        }
+    }
+
+    private boolean handlePasswordOnlyLockEditingConnections(String providedPassword, boolean dialogWasCancelled, String dialogId) {
+        boolean loadConnections;
+        if (dialogId == GetTextFragment.DIALOG_ID_GET_MATCHING_MASTER_PASSWORDS) {
```

Comment:
In java, you don't typically check equality of strings with ==. However, I don't think we should pass dialogId into this method and then check it for equality with a static string in GetTextFragment. If you wanted the method to behave differently based on whether the dialog is one vs another, you should pass into the method a boolean indicating whether it's handling a master password vs a lock settings only password.